lib/internal/crypto/keygen.js

```diff
@@ -245,6 +247,47 @@ function check(type, options, callback) {
                                             cipher, passphrase, wrap);
       }
       break;
+    case 'dh':
+      {
+        const { group, primeLength, prime, generator } = needOptions();
+        let args;
+        if (group != null) {
+          let incompatibleOption;
+          if (((incompatibleOption = 'prime') && prime != null) ||
+              ((incompatibleOption = 'primeLength') && primeLength != null) ||
+              ((incompatibleOption = 'generator') && generator != null))
+            throw new ERR_INCOMPATIBLE_OPTION_PAIR('group', incompatibleOption);
+          if (typeof group !== 'string')
+            throw new ERR_INVALID_OPT_VALUE('group', group);
+          args = [group];
+        } else {
+          if (prime != null) {
+            if (primeLength != null)
+              throw new ERR_INCOMPATIBLE_OPTION_PAIR('prime', 'primeLength');
+            if (!isArrayBufferView(prime))
+              throw new ERR_INVALID_OPT_VALUE('prime', prime);
+          } else if (primeLength != null) {
+            if (!isUint32(primeLength))
+              throw new ERR_INVALID_OPT_VALUE('primeLength', primeLength);
+          } else {
+            // TODO: Throw
+          }
+
+          if (generator != null) {
+            if (!isUint32(generator))
+              throw new ERR_INVALID_OPT_VALUE('generator', generator);
```

Comment:
Maybe use `validateUint32()` from `validators.js`? (also above)

Reply:
I don't think throwing `ERR_INVALID_ARG_TYPE` is appropriate for options. Or if it is, why do we have `ERR_INVALID_OPT_VALUE`?

Reply:
Well, that's a good question actually. Perhaps @jasnell @addaleax @Trott could chime in?
Should we maybe tweak validators to allow ERR_INVALID_OPT_VALUE? Though it is much less specific than what the validators currently throw.

Reply:
Note that the rest of the file also uses `ERR_INVALID_OPT_VALUE` for the same reason.

Reply:
I think it would be fine to add support in the necessary validators to indicate it's an option instead of an argument.

Reply:
I would prefer not to do that as part of this PR, and since the rest of the key generation API would have to be updated, too, it might make sense to do that in a follow-up semver-major change?